### ISSUE TITLE
Deserializing elements with no children or empty content doesn't work

### ISSUE THREAD

**Issue by amadfida** [FasterXML/jackson-dataformat-xml#284]
Version: 2.9.5
woodstox: 5.1.0

When deserializing empty elements such as 
```
<hostname>
</hostnames>
```
we get 

`(although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('
        ')`

our POGO looks like 

```
@Value
@JsonIgnoreProperties(ignoreUnknown = true)
public class Host {

    @Getter
    private HostNames hostnames;
```

when changed to `</hostnames>` it works - however we don't control the XML generation so need to work with whats provided. We also have  `FromXmlParser.Feature.EMPTY_ELEMENT_AS_NULL;` enabled.

**Comment by amadfida** [comment]
strangely `<hostnames></hostnames>` works ... Is there an option to trip empty string to NULL

**Comment by cowtowncoder** [comment]
I think trimming is performed by most deserializers that accept empty String; but such trimming must be handled by deserializer. So perhaps `BeanDeserializer` does not do that yet.

A unit test, with plain Java, would help here.

**Comment by cowtowncoder** [comment]
Without full test can not be 100% certain, but I think that fixes that have gone in 2.12 would have resolved this as well.
If not, may be reopened with a simple reproduction (unit test).
